ui/admin/app/components/model-providers/ModelProviderConfigure.tsx

```diff
@@ -127,7 +127,11 @@ export function ModelProviderConfigureContent({
         ModelProviderApiService.revealModelProviderById.key(modelProvider.id),
         ({ modelProviderId }) =>
             ModelProviderApiService.revealModelProviderById(modelProviderId),
-        { keepPreviousData: true }
+        {
+            keepPreviousData: true,
+            // 404 -> no credential found, so don't retry
+            shouldRetryOnError: (error) => error.status !== 404,
```

Comment:
can you do 3 things:

- add a `NotFoundError` to `apiErrors.ts` (does not need to be a BoundaryError
- add respective catch clause to `request()` in `~/lib/service/api/primitives.ts`
- replace with this:
```suggestion
            shouldRetryOnError: (error) => error instanceof NotFoundError,
```

Reply:
Oh yeah, can do! :+1: